File: SPRING_BOOT/Modern_Hospital_Management_System/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/EmergencyPatientService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import emranhss.com.Modern_Hospital_Management_System.dto.request.EmergencyPatientRequest;
import emranhss.com.Modern_Hospital_Management_System.dto.request.EmergencyStatusUpdateRequest;
import emranhss.com.Modern_Hospital_Management_System.dto.response.EmergencyDashboardResponse;
import emranhss.com.Modern_Hospital_Management_System.dto.response.EmergencyPatientResponse;

import java.util.List;

public interface EmergencyPatientService {

    EmergencyPatientResponse create(EmergencyPatientRequest request);

    EmergencyPatientResponse getById(Long id);

    List<EmergencyPatientResponse> getAll();

    List<EmergencyPatientResponse> search(String keyword);

    EmergencyPatientResponse update(Long id, EmergencyPatientRequest request);

    EmergencyPatientResponse updateStatus(Long id, EmergencyStatusUpdateRequest request);

    void delete(Long id);

    EmergencyDashboardResponse getDashboard();

    List<EmergencyPatientResponse> getByStatus(String status);

    List<EmergencyPatientResponse> getByTriageLevel(Integer triageLevel);
}

File: SPRING_BOOT/Hospital_Management_system_angular/src/app/services/emergency/emergency-patient.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { EmergencyPatient, EmergencyDashboard, EmergencyStatusUpdate } from '../../models/emergency';

@Injectable({ providedIn: 'root' })
export class EmergencyPatientService {
  private api = environment.apiUrl + 'emergency/patients';

  constructor(private http: HttpClient) {}

  create(patient: EmergencyPatient): Observable<EmergencyPatient> {
    return this.http.post<EmergencyPatient>(this.api, patient);
  }

  getAll(): Observable<EmergencyPatient[]> {
    return this.http.get<EmergencyPatient[]>(this.api);
  }

  getById(id: number): Observable<EmergencyPatient> {
    return this.http.get<EmergencyPatient>(`${this.api}/${id}`);
  }

  search(keyword: string): Observable<EmergencyPatient[]> {
    return this.http.get<EmergencyPatient[]>(`${this.api}/search?keyword=${keyword}`);
  }

  update(id: number, patient: EmergencyPatient): Observable<EmergencyPatient> {
    return this.http.put<EmergencyPatient>(`${this.api}/${id}`, patient);
  }

  updateStatus(id: number, update: EmergencyStatusUpdate): Observable<EmergencyPatient> {
    return this.http.put<EmergencyPatient>(`${this.api}/${id}/status`, update);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.api}/${id}`);
  }

  getDashboard(): Observable<EmergencyDashboard> {
    return this.http.get<EmergencyDashboard>(`${this.api}/dashboard`);
  }

  getByStatus(status: string): Observable<EmergencyPatient[]> {
    return this.http.get<EmergencyPatient[]>(`${this.api}/status/${status}`);
  }

  getByTriageLevel(level: number): Observable<EmergencyPatient[]> {
    return this.http.get<EmergencyPatient[]>(`${this.api}/triage/${level}`);
  }
}
